nuve/nuveAPI/test/mdb/tokenRegistry.js

```diff
@@ -0,0 +1,110 @@
+/*global require, describe, it, beforeEach, afterEach*/
+'use strict';
+var mocks = require('../utils');
+var sinon = require('sinon');
+var expect  = require('chai').expect;
+
+var kArbitraryToken = {_id: '1', name: 'arbitraryToken', creationDate: new Date(0)};
+var kArbitraryTokenId = '1';
+
+describe('Token Registry', function() {
+  var tokenRegistry,
+      dataBase;
+  beforeEach(function() {
+    mocks.start(mocks.licodeConfig);
+    dataBase = mocks.start(mocks.dataBase);
+    tokenRegistry = require('../../mdb/tokenRegistry.js');
+  });
+
+  afterEach(function() {
+    mocks.stop(mocks.licodeConfig);
+    mocks.stop(dataBase);
+    mocks.deleteRequireCache();
+    mocks.reset();
+  });
+
+  it('should return a list of tokens when getList is called', function() {
+    dataBase.db.tokens.find.returns({
+      toArray: function(cb) {
+        cb(null, [kArbitraryToken]);
+      }
+    });
+    var callback = sinon.stub();
+    tokenRegistry.getList(callback);
+
+    expect(callback.calledWith([kArbitraryToken])).to.be.true;  // jshint ignore:line
+  });
+
+  it('should return undefined if not found in the database when getToken is called', function() {
+    var callback = sinon.stub();
+    dataBase.db.tokens.findOne.callsArgWith(1, null, undefined);
+    tokenRegistry.getToken(kArbitraryTokenId, callback);
+
+    expect(callback.calledWith(undefined)).to.be.true;  // jshint ignore:line
+  });
+
+  it('should return a token from the database when getToken is called', function() {
+    var callback = sinon.stub();
+    dataBase.db.tokens.findOne.callsArgWith(1, null, kArbitraryToken);
+    tokenRegistry.getToken(kArbitraryTokenId, callback);
+
+    expect(callback.calledWith(kArbitraryToken)).to.be.true;  // jshint ignore:line
+  });
+
+  it('should return false if the token is not found when hasToken is called', function() {
+    var callback = sinon.stub();
+    dataBase.db.tokens.findOne.callsArgWith(1, null, undefined);
+    tokenRegistry.hasToken(kArbitraryTokenId, callback);
+
+    expect(callback.calledWith(false)).to.be.true;  // jshint ignore:line
+  });
+
+  it('should return true if the token is found in database when hasToken is called', function() {
+    var callback = sinon.spy();
```

Comment:
Consider using either spy or stub consistently. I see nothing wrong with using stubs where spies would suffice but randomly using one or the other creates confusion